### DISCUSSION TITLE
Rook-ceph spams logs

### DISCUSSION THREAD

**Discussion by Shaban-Mahmood** [rook/rook#14259] [General]
Hi,
We are facing challenges with rook-ceph spamming our logs with many unnecessary logs, especially debug logs. We have observed that most logs originate from rook-ceph-mon or rook-ceph-mgr.

We would like to limit these logs and have already tried to follow your guidance: https://docs.ceph.com/en/reef/rados/troubleshooting/log-and-debug/ regarding your Debug logging and subsystems. We have limited logs from all subsystems, which has of course helped but not significantly.

Here are examples of the logs we are receiving, which we would like to limit:
cluster [DBG] pgmap v988: 169 pgs: 169 active+clean; 15 GiB data, 31 GiB used, 23 TiB / 23 TiB avail; 1.1 KiB/s rd, 456 KiB/s wr, 43 op/s

audit 2024-05-17T09:14:53.900401+0000 mon.a (mon.0) 2296 : audit [DBG] from='admin socket' entity='admin socket' cmd='mon_status' args=[]: dispatch

We need assistance in limiting these logs. Can you help us with that?

**Comment by travisn** [comment]
A couple of ideas:
- Change the Ceph component log levels, but sounds like you've already done that 
- Rotate the logs with an increased `periodicity` or reduced `maxLogSize` as shown [here](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L137-L140)

**Reply by Shaban-Mahmood** [reply]
Hi,
Thank you for the suggestions. However, we are already running with those configurations but still have the issue with many logs. 
Any other suggestions?

**Reply by travisn** [reply]
@Shaban-Mahmood Are you concerned that the logs are large because of those two log entries? How verbose are they? Are they causing the logs to be rotated much sooner than you prefer? I'm trying to understand the real motivation and severity for why these need to be suppressed.

**Reply by Shaban-Mahmood** [reply]
Sorry for the late reply. We have a lot of log entries from rook-ceph and they are spamming our logs, making it hard to see relevant logs. Especially just after our system have deployed, it spams the log and I have seen 80.000 and 150.000 log entries within 5 min after deploying our system and it continues.

**Comment by GabMgt** [comment]
Same problem here. Did you manage to fix the issue?

**Reply by subhamkrai** [reply]
@GabMgt what version of rook you are using?

**Reply by GabMgt** [reply]
v1.5.5

**Reply by subhamkrai** [reply]
can you try to upgrade to latest version to see if those are gone? v1.5.5 is pretty old

**Reply by GabMgt** [reply]
I'll try, but I can't guarantee anything
I wasn't the one who set up Rook on this cluster.

**Reply by travisn** [reply]
Make sure to review the [upgrade guide](https://rook.io/docs/rook/latest/Upgrade/rook-upgrade/) to upgrade between each release.

**Reply by GabMgt** [reply]
I did the upgrade and the cluster is currently running the v1.15.3 version
I can see the same logs sadly

**Reply by travisn** [reply]
@GabMgt Did you try the workaround [here](https://github.com/rook/rook/discussions/14259#discussioncomment-10697971)?

**Comment by harveyjing** [comment]
<img width="305" alt="image" src="https://github.com/user-attachments/assets/a2c0fbcd-1e51-4c53-84d7-f84a99bad921">
Same issue. This is the statistics of elasticsearch. I am using 1.14. Did it fix in 1.15?

**Reply by travisn** [reply]
What is the meaning of the percent value?

**Reply by harveyjing** [reply]
Log lines statistics of last 15 minutes. There are a lot of debug level logs, but I didn't change any default log level (the default level in operator helm values is INFO).
Just like this:
> Here are examples of the logs we are receiving, which we would like to limit:
cluster [DBG] pgmap v988: 169 pgs: 169 active+clean; 15 GiB data, 31 GiB used, 23 TiB / 23 TiB avail; 1.1 KiB/s rd, 456 KiB/s wr, 43 op/s
>audit 2024-05-17T09:14:53.900401+0000 mon.a (mon.0) 2296 : audit [DBG] from='admin socket' entity='admin socket' cmd='mon_status' args=[]: dispatch

**Reply by travisn** [reply]
One workaround for any log entry with these prefixes can be filtered out completely with the following toolbox command for each daemon where you want to apply it:
```
# suppress logs with the prefix "debug"
ceph config set mon.a "log to stderr" false

# suppress logs with the prefix "cluster" and "audit"
ceph config set mon.a "mon cluster log to stderr" false
```
However, this may hide too much in case troubleshooting is needed. But does this help?

**Reply by harveyjing** [reply]
Thank you so much for those clues. I searched the web and found the [documentation](https://docs.ceph.com/en/octopus/cephadm/operations/):
>ceph config set global log_to_stderr false
ceph config set global mon_cluster_log_to_stderr false

It seems that it's not a Rook issue but a Ceph one.
It worked and thank you again.

**Reply by W1zzardTPU** [reply]
> ```
> # suppress logs with the prefix "debug"
> ceph config set mon.a "log to stderr" false
> 
> # suppress logs with the prefix "cluster" and "audit"
> ceph config set mon.a "mon cluster log to stderr" false
> ```

Do these commands really do what they claim to do based on the comment?

The help texts for these options say:
log_to_stderr -> "send log lines to stderr"
mon_cluster_log_to_stderr -> "Make monitor send cluster log messages to stderr (prefixed by channel)"
